### PR TITLE
[plg_content_pagebreak] Let users template override TOC

### PR DESCRIPTION
**Pull request by ReLater**
See discussion here: https://github.com/joomla/joomla-cms/pull/17670

### Summary of Changes
- Added possibility to use a template override for TOC.
- Don't break B\C.
- BTW: Fix older bug: "Show All" never has `active` class even if active.

### Testing Instructions
- Plugin "Content - Page Break". Make settings:

- Add some pagebreak markers to an article and open article in frontend.
- Inspect TOC (also source code).

- Apply patch.
- Inspect TOC (also source code).

- Expected: No changes.

- Create a template override for new file tmpl/toc.php.
Example with Protostar:

- Check that the override works.

### CODE REVIEW

**Comment by dgrammatiko** [comment]
we need to make sure that all plugins are actually overridable! Great work here 👍

File: plugins/content/pagebreak/tmpl/toc.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+defined('_JEXEC') or die;
+?>
+<div class="pull-right article-index">
+
+	<?php if ($headingtext) : ?>
+	<h3><?php echo $headingtext; ?></h3>
+	<?php endif; ?>
+
+	<ul class="nav nav-tabs nav-stacked">
+	<?php foreach ($list as $listItem) :
```
Comment (Quy) [inline plugins/content/pagebreak/tmpl/toc.php]: Change to per [Joomla's Coding Standards](user-content-mixed-language-usage-eg-at-the-layout-files).

```
<?php foreach ($list as $listItem) : ?>
        <?php $class = $listItem->liClass ? ' class="' . $listItem->liClass . '"' : ''; ?>
```

File: plugins/content/pagebreak/pagebreak.php [hunk 2]
```diff
@@ -310,22 +302,28 @@ protected function _createToc(&$row, &$matches, &$page)
...
-				. JText::_('PLG_CONTENT_PAGEBREAK_ALL_PAGES') . '</a></li>';
+			$list[$i]          = new stdClass;
+			$list[$i]->link    = JRoute::_(ContentHelperRoute::getArticleRoute($row->slug, $row->catid, $row->language) . '&showall=1&limitstart=');
+			$list[$i]->liClass = ($showall === 1) ? 'active' : '';
+			$list[$i]->class   = ($showall === 1) ? 'toclink active' : 'toclink';
+			$list[$i]->title   = JText::_('PLG_CONTENT_PAGEBREAK_ALL_PAGES');
 		}
-
-		$row->toc .= '</ul></div>';
+
```
Comment (Quy) [inline plugins/content/pagebreak/pagebreak.php]: Remove tabs

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 50c8e5174cfbb4f4e9d4403679379533e3ee8e26<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20202">issues.joomla.org/tracker/joomla-cms/20202</a>.</sub>

**Review by LivioCavallo — APPROVED** [review]

**Comment by LivioCavallo** [comment]
I have tested this item :white_check_mark: successfully on 50c8e5174cfbb4f4e9d4403679379533e3ee8e26<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20202">issues.joomla.org/tracker/joomla-cms/20202</a>.</sub>

**Comment by LivioCavallo** [comment]
Well done. Thanks. I tested it successfully. If all will be ok I'll close #17670

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20202">issues.joomla.org/tracker/joomla-cms/20202</a>.</sub>

**Pull request merged**